### DISCUSSION TITLE
Add first class sentry support

### DISCUSSION THREAD

**Discussion by zschumacher** [PrefectHQ/prefect#4672] [Ideas]
It would be amazing if prefect had first class support for sentry.  This could be as simple as a flow automatically looking for SENTRY_DSN and SENTRY_ENVIRONMENT in secrets or the environment if the `sentry_sdk` extra is installed.

**Comment by zschumacher** [comment]
FWIW im happy to take a stab at a PR for this as well, just wanted opinions on if others agree its a good idea.

**Reply by zanieb** [reply]
Hey @zschumacher -- it may be helpful to explain how sentry would integrate with Prefect and what the value-add would be for those of us who are not familiar with it :)

**Reply by zschumacher** [reply]
Sure thing @madkinsz.  sentry works by calling `sentry_sdk.init()` in the entry point of your code.  I would assume this would live in the `Runner` entry point, but I'm not sure what is best.  

Sentry is a saas that captures any uncaught exceptions and stores your traceback.  For the most part, it really __just works__.  The only configuration necessary is passing `sentry_dsn` and `sentry_environment` to the sdk init func.  It falls back to looking at `SENTRY_DSN` and `SENTRY_ENVIRONMENT` env vars if those kwargs aren't passed.

Airflow has first class sentry support, and I think it'd be a great addition to the lib.  Relevant links below:

home page - https://sentry.io
python sdk - https://docs.sentry.io/platforms/python/
airflow docs - https://docs.sentry.io/platforms/python/guides/airflow/
airflows gross implementation - https://github.com/apache/airflow/blob/main/airflow/sentry.py

**Reply by zanieb** [reply]
Thanks for the additional details and the reminder ping :) I'll chat with the team about this. I'm not sure we'd do something like `sentry_sdk.init()` because the runner shouldn't really have any uncaught exceptions. We'd want to tie this into task states which feels more complicated.

**Reply by mikicz** [reply]
Hey, did y'all have that discussion? This feature would be really useful!

**Reply by abhishekmishragithub** [reply]
Hello folks, this feature would be very useful.
@madkinsz are you folks planning to include this in upcoming releases?

**Reply by tonal** [reply]
How to about Prefect 2 integrates with sentry?

**Comment by mattiamatrix** [comment]
Hello! It would be great to have a Sentry integration in Prefect. Are there any updates on this topic?

**Comment by tonal** [comment]
How to about Prefect 2 and 3 integrates with sentry?

**Comment by tonal** [comment]
How to about Prefect 3 integrates with sentry?

**Reply by abhishekmishragithub** [reply]
you are not losing hope @tonal 💪🏼
